jflex/src/main/java/jflex/GeneratorOptions.java

```diff
@@ -0,0 +1,92 @@
+package jflex;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Optional;
```

Comment:
AutoValue is only an annotation with compile-time dependency. In the end, this automated approach will only bring one (generated) class in the jar. I think its brings enough value (cleaner definition compared to getters and setters ; immmutable object by nature ; free toString and free equals) that's I want to convince you it's worth it.

On the other hand, guava is a huge beast. I didn't think about the cost, because I'm used to use it. It's probably not woth it just for Optional.

Reply:
I don't actually find it cleaner ;-) It probably is cleaner when you are used to the framework, because then it's a standardised pattern, but if you're not used to it, it's another largish library to look at.

Reply:
On the dependency point, separately, because I feel somewhat strongly about that:

Introducing additional dependencies can have a rather large cost, it depends on how large the project is and what the dependencies are doing. Mostly, these are:
 * you need to understand the dependency/library you are using
 * you need to deal with the additional complexity of managing them. mvn does that well, but that's only one of many build methods we are trying to support.
 * users (developers) need to track and install them as well
 * the dependencies will develop and you have to maintain compatibility

We've so far managed to have the cup runtime as the only dependency for JFlex, which is very nice.

Of course there are many advantages as well, esp if you are using those dependencies a lot or know them well anyway. It's a trade-off. The simple kind of CLI parsing and options would be worth a dependency if we had a set of projects that all do the same thing. For just JFlex, that code is so simple that I find it much easier to avoid the build time messiness and the need to keep track of what an external library is doing.